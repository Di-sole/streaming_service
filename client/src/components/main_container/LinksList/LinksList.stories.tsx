import { Meta, StoryObj } from '@storybook/react';
import './LinksList.module.scss';
import LinksList from './LinksList';

const meta: Meta<typeof LinksList> = {
    title: 'Links/LinksList',
    component: LinksList,
    tags: ['autodocs'],
    decorators: [
        (Story) => (
            <div style={{ background: '#100e19', padding: '10px' }}>
                <Story />
            </div>
        ),
    ],
};

export default meta;

type Story = StoryObj<typeof LinksList>;

export const Row: Story = {
    args: {
        direction: 'row',
        links: [
            {
                href: '/',
                text: 'Ссылка 1',
            },
            {
                href: '/',
                text: 'Ссылка 2',
            },
            {
                href: '/',
                text: 'Ссылка 3',
            },
            {
                href: '/',
                text: 'Ссылка 4',
            },
            {
                href: '/',
                text: 'Ссылка 5',
            },
        ],
    },
};

export const Column: Story = {
    args: {
        direction: 'column',
        links: [
            {
                href: '/',
                text: 'Ссылка 1',
            },
            {
                href: '/',
                text: 'Ссылка 2',
            },
            {
                href: '/',
                text: 'Ссылка 3',
            },
            {
                href: '/',
                text: 'Ссылка 4',
            },
            {
                href: '/',
                text: 'Ссылка 5',
            },
        ],
    },
};
