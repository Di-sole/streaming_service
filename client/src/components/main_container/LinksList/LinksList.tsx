import styles from './LinksList.module.scss';
import TextLinkUI from '../../UI/links/TextLink/TextLinkUI';

interface ILink {
    href: string;
    text: string;
    option?: 'bright' | 'dim' | 'gradient';
    className?: string;
    onMouseOver?: () => void;
}

interface LinksListProps {
    direction: 'row' | 'column';
    links: ILink[];
}

const LinksList = ({ links, direction }: LinksListProps) => {
    return (
        <ul
            className={[direction === 'row' ? styles.container : styles.container__column, styles.container__menu].join(
                ' ',
            )}
        >
            {links.map((link) => (
                <li className={styles.container__item} key={link.text}>
                    <TextLinkUI
                        className={link.className}
                        onMouseOver={() => (link.onMouseOver ? link.onMouseOver() : '')}
                        href={link.href}
                        option={link.option ?? 'dim'}
                    >
                        {link.text}
                    </TextLinkUI>
                </li>
            ))}
        </ul>
    );
};

export default LinksList;
