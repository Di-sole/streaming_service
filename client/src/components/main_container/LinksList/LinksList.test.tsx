import { render, screen } from '@testing-library/react';
import LinksList from './LinksList';

const links = [
    {
        href: '/',
        text: 'Ссылка 1',
    },
    {
        href: '/',
        text: 'Ссылка 2',
    },
    {
        href: '/',
        text: 'Ссылка 3',
    },
    {
        href: '/',
        text: 'Ссылка 4',
    },
    {
        href: '/',
        text: 'Ссылка 5',
    },
];

describe('LINKSLIST SNAPSHOTS TESTS', () => {
    test('direction row LinksList should not be changed', () => {
        render(<LinksList direction="row" links={links} />);

        const list = screen.getByRole('list');
        expect(list).toMatchSnapshot();
    });

    test('direction column LinksList should not be changed', () => {
        render(<LinksList direction="column" links={links} />);

        const list = screen.getByRole('list');
        expect(list).toMatchSnapshot();
    });
});
