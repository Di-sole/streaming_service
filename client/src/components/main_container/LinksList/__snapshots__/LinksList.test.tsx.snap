// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LINKSLIST SNAPSHOTS TESTS direction column LinksList should not be changed 1`] = `
<ul
  class="container__column container__menu"
>
  <li
    class="container__item"
  >
    <a
      class="container container_dim"
      href="/"
    >
      Ссылка 1
    </a>
  </li>
  <li
    class="container__item"
  >
    <a
      class="container container_dim"
      href="/"
    >
      Ссылка 2
    </a>
  </li>
  <li
    class="container__item"
  >
    <a
      class="container container_dim"
      href="/"
    >
      Ссылка 3
    </a>
  </li>
  <li
    class="container__item"
  >
    <a
      class="container container_dim"
      href="/"
    >
      Ссылка 4
    </a>
  </li>
  <li
    class="container__item"
  >
    <a
      class="container container_dim"
      href="/"
    >
      Ссылка 5
    </a>
  </li>
</ul>
`;

exports[`LINKSLIST SNAPSHOTS TESTS direction row LinksList should not be changed 1`] = `
<ul
  class="container container__menu"
>
  <li
    class="container__item"
  >
    <a
      class="container container_dim"
      href="/"
    >
      Ссылка 1
    </a>
  </li>
  <li
    class="container__item"
  >
    <a
      class="container container_dim"
      href="/"
    >
      Ссылка 2
    </a>
  </li>
  <li
    class="container__item"
  >
    <a
      class="container container_dim"
      href="/"
    >
      Ссылка 3
    </a>
  </li>
  <li
    class="container__item"
  >
    <a
      class="container container_dim"
      href="/"
    >
      Ссылка 4
    </a>
  </li>
  <li
    class="container__item"
  >
    <a
      class="container container_dim"
      href="/"
    >
      Ссылка 5
    </a>
  </li>
</ul>
`;
